yt_dlp/extractor/bilibili.py

```diff
@@ -746,16 +746,29 @@ def _real_extract(self, url):
 
 
 class BiliBiliPlayerIE(InfoExtractor):
-    _VALID_URL = r'https?://player\.bilibili\.com/player\.html\?.*?\baid=(?P<id>\d+)'
-    _TEST = {
+    _VALID_URL = r'https?://player\.bilibili\.com/player\.html\?.*?\b(aid=(?P<id>\d+)|bvid=(?P<id_bv>[^/?#&=]+))'
+    _TESTS = [{
         'url': 'http://player.bilibili.com/player.html?aid=92494333&cid=157926707&page=1',
         'only_matching': True,
-    }
+    }, {
+        'url': 'https://player.bilibili.com/player.html?aid=996153&bvid=BV1bs411Z7nC&cid=1440287&page=1',
+        'only_matching': True,
+    }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        u = compat_urllib_parse_urlparse(url)
+        query = compat_parse_qs(u.query)
+        if "bvid" in query:
```

Comment:
do no use this.
U are already capturing this info in the regex. Leverage that.

Reply:
Parse URL query using regex is clumsy and less robust. To capture both aid and bvid, it would be like
```
    _VALID_URL = r'https?://player\.bilibili\.com/player\.html\?(?:.*?\baid=(?P<id>\d+))?(?:.*?\bbvid=(?P<id_bv>[^/?#&=]+))?(?:.*?\bpage=(?P<page>\d+))?'
```
Even that, the order of aid, bvid and page need to be fixed. 

Using standard url parser is much clean.